### DISCUSSION TITLE
Instagram app with internal proxy system اینستاگرام ضد فیلتر با پروکسی داخلی

### DISCUSSION THREAD

**Discussion by its-mx** [WeAreMahsaAmini/Core#455] [Ideas]
نسخه ی فارسی این متن را در پایین نوشته ام

Hey!
I can embed internal proxy mechanism in Instagram app and build a new client (let's call it IRGram).
What's benefit:

1) IRGram will have internal proxy mechanism like what Telegarm has. Iranian people can use it without VPN, just they need to change their proxy inside the app if their previous proxy get filtered. We (or any volunteer) will provide them new available proxies internally via push notification and other mechanisms.

2) Anyone who provide proxy servers will make money by IRGram allow them to show a-not-annoying-advertising in a dedicated place inside the app while the user is using their proxy. Why we do this? This will support the proxy servers owner and let them to keep their proxy system up and add new proxies over the time. The fight against filtering system can powerfully keep going!

3) We will track the filtering system and by the time, we can find out which IRGram's clients are imposter! After detect them, they won't receive new proxies update, so our proxies won't get banned/filtered anymore!

4) By the time, we will have a valid IRGram community cleared from regime imposters!
And IRGram will activate a new mechanism to send secure messages over Instagram servers which is only readable by other IRGram clients! Imposters won't know where and what time the next protest would be:)

5) By getting permission from the users, IRGram will do an automated group report for regime and their lovers account! By having a large IRGram community, regime accounts would be removed from Instagram fast af!

6) We can fight against Meta censorship... let's talk about it later;)
***------------------------------------------------***
Is it really possible? Yes! I have already made similar IRGram android app long time ago for myself:)
I just need to add internal proxy system and some other stuff.
Can IRGram update with each update of Instagram? New Instagram versions are using "google play protect" to stop developer making IRGram similar apps, but theoretically I can bypass the protect system:) but it would need a little money for testing that in real environment (about 3k$). But apart from this, my old app(I made it a year ago) is still working with no problem.
Can instagram detect IRGram and remove it or stop services for it? No! IRGram is exactly the original instagram client! Even Instagram servers can't detect it's another client:)
***-------------------------------------------------***
Completing this project need some investment(about 5k$)
Can you invest on it? Or can you find some fund for this project?
And Do you have any other suggestion about what else we can do in IRGram? 
Let me know!
-M4-

سلام
من میتونم مکانیسم پروکسی داخلی رو توی اپلیکیشن اینستاگرام قرار بدم (اسمش اپ رو بزاریم IRGram)
این کار چه فایده ای داره؟
۱) دقیقا مثل مکانیسمی که الان تلگرام داره، کاربرای IRGram دیگه لازم نیست vpn روشن کنن، فقط کافیه اگه پروکسی قبلی فیلتر شده، یه پروکسی جدید رو از تو لیست پروکسی های داخل اپلیکیشن فعال کنن. ما هر لحظه پروکسی های جدید رو با مکانیسم پوش نوتیفیکیشن و دیگر مکانیسم ها برای کاربران ارسال میکنیم.

۲) هرکسی که پروکسی سرور ارائه میکنه، امکان این رو داره تا زمانی که کاربر ازون پروکسی داره استفاده میکنه، به کاربران در یک قسمت مشخص از اپلیکیشن(فرض کنید اولین پست یا اولین استوری کاربر) تبلیغ نشون بده. این کار باعث میشه کسایی که پروکسی سرور ارائه میکنند بتونن هزینه های سرور هاشون رو در بیارن و مدام بتونن تعداد بیشتری پروکسی ارائه کنن، و همچنین باعث ایجاد انگیزه برای اینکه تعداد آدم های بیشتری بیان پروکسی سرور ارائه کنن خواهد شد.

۳) اپلیکیشن IRGram نحوه ی فیلتر شدن و بن شدن پروکسی ها رو رصد خواهد کرد و ازین طریق به مرور زمان میتونه بفهمه کدوم کلاینت ها دست این متخصص های بی شرف فیلتر کننده هست و دیگه برای اونها پروکسی های جدید رو نمیفرسته، و در نتیجه کم کم سیستم فیلترین لنگ در هوا میمونه و آدرس سرور پروکسی های جدید دیگه به راحتی رد یابی و فیلتر نمیشن:)

۴) به مرور زمان با توجه به مورد 3، کسایی که از IGRam استفاده میکنند خودی ها میشن و نخودی ها از اپلیکیشن بن میشن، و بعدش یه سیستم جدید ارسال پست و استوری راه میندازیم که فقط کاربرای IRGram میتونن اون پست و استوری رو ببینند(راستی، IRGram رو همون شبکه اینستاگرام اصلی کار میکنه ها! اصلا همون اینستاگرام اصلیه هیچ فرقی نداره فقط یسری کد بهش اضافه شده) و با این امکان موزدور ها دیگه نمیتونن بفهمند تظاهرات بعدی چه ساعتی و کجا خواهد بود:)

۵) با اجازه ی یوزرها، یه سیستم ریپورت اتوماتیک گروهی راه میندازیم که اکانت های ج ا رو ریپورت میکنه. هرچی کامیونتی تعدادش بیشتر بشه زودتر مزدورها رو از اینستاگرام حذف میکنیم

۶) ما حتی میتونی با شرکت متا که داره محتپا هارو ساسنور میکنه و اکانت ها رو میبنده هم وارد جنگ بشیم:) اینو بزارین واسه بعد فعلا

***-------------------------------------------***
آیا واقعا امکان پذیره؟ آره! من قبلا همچین چیزیو واسه خودم نوشتم(واسه اندروید) فقط نیازه سیستم پروکسی رو بهش اضافه کنم و یسری تغییرات بدم. چیزی که دارم قابل ارائه و استفاده هست(خودم استفاده میکنم هر روز)

آیا IRGram با هر ورژن جدید که برای اینستاگرام میاد میتونه آپدیت بشه؟ ببینید در ورژن های جدید اینستاگرام، متا اومده واسه جلوگیری از ساخت همچین کلاینت هایی، از سرویس جدید google play protect تو اپلیکیشن اینستاگرام استفاده کرده. به لحاظ تئوری من همچنان میتونم این سیستم رو دور بزنم:) ولی باید تست واقعی کنم، و تست اش هم هزینه‌بر هست(2 یا 3 هزار دلاری لازمه حداقل). ولی حدای ازین، همچنان اپلیکیشن قدیمیه که ساختم راحت بدون مشکل کار میکنه(منطبق با نسخه یکسال پیش اینستاگرامه).

اینستاگرام میتونه IRGram رو تشخیص بده و سرور هاشو قطع کنه براش؟(چون مستقیم عین کلاینت اصلی به api های private وصلیم)
نه به هیچ وجه:) سرور های اینستاگرام حتی نمیتونن تشخیص بدن ما IRGram هستیم و اپ اوریجینال خود اینستا نیستیم چه برسه‌به بقیه موارد:)
***------------------------------------------***
تکمیل این کار نیاز به حمایت مالی داره(حدود 5 هزار دلار)
میتونی برای ساخت اپ حمایت مالی کنی یا فاندی جذب کنی؟ 
و اینکه آیا ایده دیگه ای داری که‌ به اپ اضافه بشه؟
 برام بنویس!

**Comment by fitzli5** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Reply by Dorood-bar-shoma** [reply]
میدونم داری زحمت میکشی. ولی اگه قراره ما هم استفاده کنیم پس لطفاً به زبان ساده تر بگو چیکار باید بکنیم
